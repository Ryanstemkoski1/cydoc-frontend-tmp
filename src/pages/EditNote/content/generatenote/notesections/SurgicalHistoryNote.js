import React from 'react';
import { Table } from 'semantic-ui-react';

class SurgicalHistoryNote extends React.Component {
    render() {
        const surgicalHistory = this.props.surgicalHistory;

        if (this.props.isRich) {
            return (
                <Table>
                    <Table.Header>
                        <Table.Row>
                            <Table.HeaderCell>Procedure</Table.HeaderCell>
                            <Table.HeaderCell>Date</Table.HeaderCell>
                            <Table.HeaderCell>Comments</Table.HeaderCell>
                        </Table.Row>
                    </Table.Header>
                    <Table.Body>
                    {Object.keys(surgicalHistory).map(key => (
                        surgicalHistory[key].Procedure !== "" ?
                        <Table.Row>
                            <Table.Cell>{surgicalHistory[key].Procedure}</Table.Cell>
                            <Table.Cell>{surgicalHistory[key].Date ? surgicalHistory[key].Date : null}</Table.Cell>
                            <Table.Cell>{surgicalHistory[key].Comments ? surgicalHistory[key].Comments : null}</Table.Cell>
                        </Table.Row> : null
                    ))}
                    </Table.Body>
                </Table>
            )
        }

        return (
            <ul>
                {Object.keys(surgicalHistory).map(key => (
                    surgicalHistory[key].Procedure !== "" ?
                    <li>
                        <b>{surgicalHistory[key].Procedure} </b>
                        {surgicalHistory[key].Date ? `${surgicalHistory[key].Date}. ` : null}
                        {surgicalHistory[key].Comments ? surgicalHistory[key].Comments : null}
                    </li> : null
                ))}
            </ul>
        )
    }
}

export default SurgicalHistoryNote;