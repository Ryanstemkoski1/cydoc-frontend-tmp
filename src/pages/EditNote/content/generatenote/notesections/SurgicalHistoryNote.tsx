import React, { Component } from 'react';
import { SurgicalHistoryState } from 'redux/reducers/surgicalHistoryReducer';
import { Table } from 'semantic-ui-react';

interface SurgicalHistoryProps {
    isRich: boolean;
    surgicalHistory: SurgicalHistoryState;
}

export class SurgicalHistoryNote extends Component<SurgicalHistoryProps> {
    checkEmpty = () => {
        for (const id in this.props.surgicalHistory) {
            if (
                !(
                    this.props.surgicalHistory[id].procedure === '' &&
                    this.props.surgicalHistory[id].year === -1 &&
                    this.props.surgicalHistory[id].comments === ''
                )
            ) {
                return false;
            }
        }
        return true;
    };

    trimSurgicalProcedure = (procedure: string) => {
        return procedure.split(' ').slice(1).join(' ');
    };

    render() {
        const { isRich, surgicalHistory } = this.props;

        if (this.checkEmpty()) {
            return <div />;
        } else if (isRich) {
            return (
                <Table>
                    <Table.Header>
                        <Table.Row>
                            <Table.HeaderCell>Procedure</Table.HeaderCell>
                            <Table.HeaderCell>Year</Table.HeaderCell>
                            <Table.HeaderCell>Comments</Table.HeaderCell>
                        </Table.Row>
                    </Table.Header>
                    <Table.Body>
                        {Object.values(surgicalHistory).map((surgical) =>
                            surgical.procedure !== '' ? (
                                <Table.Row>
                                    <Table.Cell>
                                        {this.trimSurgicalProcedure(
                                            surgical.procedure
                                        )}
                                    </Table.Cell>
                                    <Table.Cell>
                                        {surgical.year !== -1
                                            ? surgical.year
                                            : null}
                                    </Table.Cell>
                                    <Table.Cell>
                                        {surgical.comments
                                            ? surgical.comments
                                            : null}
                                    </Table.Cell>
                                </Table.Row>
                            ) : null
                        )}
                    </Table.Body>
                </Table>
            );
        } else {
            return (
                <ul>
                    {Object.values(surgicalHistory).map((surgical) =>
                        surgical.procedure !== '' ? (
                            <li>
                                <b>
                                    {this.trimSurgicalProcedure(
                                        surgical.procedure
                                    )}{' '}
                                </b>
                                {surgical.year !== -1
                                    ? `${surgical.year}. `
                                    : null}
                                {surgical.comments ? surgical.comments : null}
                            </li>
                        ) : null
                    )}
                </ul>
            );
        }
    }
}

export default SurgicalHistoryNote;
