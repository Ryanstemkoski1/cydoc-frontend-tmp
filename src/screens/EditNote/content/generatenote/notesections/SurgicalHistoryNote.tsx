import { YesNoResponse } from '@constants/enums';
import React, { Component } from 'react';
import { SurgicalHistoryState } from '@redux/reducers/surgicalHistoryReducer';
import { Table } from 'semantic-ui-react';

interface SurgicalHistoryProps {
    isRich: boolean;
    surgicalHistory: SurgicalHistoryState;
}

export class SurgicalHistoryNote extends Component<SurgicalHistoryProps> {
    checkEmpty = () => {
        const surgicalHistory = this.props.surgicalHistory.elements;
        for (const id in surgicalHistory) {
            if (surgicalHistory[id].hasHadSurgery != YesNoResponse.None) {
                return false;
            }
        }
        return true;
    };

    trimSurgicalProcedure = (procedure: string) => {
        return procedure;
    };

    render() {
        const { isRich } = this.props;
        const { hasSurgicalHistory } = this.props.surgicalHistory;
        const surgicalHistory = this.props.surgicalHistory.elements;

        // hasSurgicalHistory is value of the YES/NO question on surgical history form
        if (hasSurgicalHistory === false) {
            return <div>The patient has never had any surgeries.</div>;
        } else if (this.checkEmpty()) {
            return <div />;
        } else if (isRich) {
            return (
                <Table>
                    <Table.Header>
                        <Table.Row>
                            <Table.HeaderCell>Procedure</Table.HeaderCell>
                            <Table.HeaderCell>Year</Table.HeaderCell>
                            <Table.HeaderCell>Comments</Table.HeaderCell>
                        </Table.Row>
                    </Table.Header>
                    <Table.Body>
                        {Object.values(surgicalHistory).map(
                            (surgical, index) =>
                                surgical.procedure !== '' ? (
                                    <Table.Row key={index}>
                                        <Table.Cell>
                                            {this.trimSurgicalProcedure(
                                                surgical.procedure
                                            )}
                                        </Table.Cell>
                                        <Table.Cell>
                                            {surgical.year !== -1
                                                ? surgical.year
                                                : null}
                                        </Table.Cell>
                                        <Table.Cell>
                                            {surgical.comments
                                                ? surgical.comments
                                                : null}
                                        </Table.Cell>
                                    </Table.Row>
                                ) : null
                        )}
                    </Table.Body>
                </Table>
            );
        } else {
            return (
                <ul>
                    {Object.values(surgicalHistory).map((surgical, index) =>
                        surgical.procedure !== '' ? (
                            <li key={index}>
                                <b>
                                    {this.trimSurgicalProcedure(
                                        surgical.procedure
                                    )}{' '}
                                </b>
                                {surgical.year !== -1
                                    ? `${surgical.year}. `
                                    : null}
                                {surgical.comments ? surgical.comments : null}
                            </li>
                        ) : null
                    )}
                </ul>
            );
        }
    }
}

export default SurgicalHistoryNote;
