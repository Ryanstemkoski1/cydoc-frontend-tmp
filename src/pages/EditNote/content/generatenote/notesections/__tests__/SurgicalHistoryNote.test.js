import React from 'react';
import Enzyme, { mount } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import { SurgicalHistoryNote } from '../SurgicalHistoryNote';

Enzyme.configure({ adapter: new EnzymeAdapter() });

const surgical = {
    foo: { procedure: '', year: -1, comments: '' },
    bar: { procedure: '', year: -1, comments: '' },
};

const mountWithProps = (surgicalHistory = surgical, isRich = false) => {
    return mount(
        <SurgicalHistoryNote
            surgicalHistory={surgicalHistory}
            isRich={isRich}
        />
    );
};

describe('Surgical History Note', () => {
    const nonEmptySH = {
        empty: { procedure: '', year: 1000, comments: 'bar' },
        foo: { procedure: '1 aaa', year: -1, comments: 'xxx' },
        test: { procedure: '2 bbb', year: 2000, comments: '' },
    };

    it('renders without crashing', () => {
        const wrapper = mountWithProps();
        expect(wrapper).toBeTruthy();
    });

    it('matches snapshot', () => {
        const wrapper = mountWithProps();
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders correctly when completely empty', () => {
        const wrapper = mountWithProps();
        expect(wrapper.text()).toContain('');
    });

    // // TODO: Fix below tests
    // it('renders only non-empty entries (non-rich)', () => {
    //     const wrapper = mountWithProps({ ...nonEmptySH });
    //     expect(wrapper.find('li')).toHaveLength(2);
    //     expect(wrapper.text()).toContain('aaa xxx'); // don't show year
    //     expect(wrapper.text()).toContain('bbb 2000. '); // don't show comment
    //     expect(wrapper.text()).not.toContain('1000. bar');
    // });

    // it('renders only non-empty entries (rich-text)', () => {
    //     const wrapper = mountWithProps({ ...nonEmptySH }, true);
    //     expect(wrapper.find('table')).toHaveLength(1);
    //     expect(wrapper.find('tr')).toHaveLength(3);
    //     const expectedCells = [
    //         ['Procedure', 'Year', 'Comments'],
    //         ['aaa', '', 'xxx'],
    //         ['bbb', '2000', ''],
    //     ];
    //     wrapper
    //         .find('tr')
    //         .forEach((row, r) =>
    //             row
    //                 .children()
    //                 .forEach((node, idx) =>
    //                     expect(node.text()).toEqual(expectedCells[r][idx])
    //                 )
    //         );
    // });
});
