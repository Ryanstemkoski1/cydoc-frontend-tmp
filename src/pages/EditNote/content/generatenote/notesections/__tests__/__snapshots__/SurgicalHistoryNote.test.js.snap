// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Surgical History Note matches snapshot 1`] = `"<div>No surgical history reported.</div>"`;
